internal/indexer/data/mongo_client.py

```
@@ -0,0 +1,205 @@
+        collection = self.db[METADATA_COLLECTION]
+        result = collection.find_one(
+            {"_id": normalized_url},
+        )
+
+        return Metadata.from_dict(result)
+
+    def create_metadata_entry_operation(
+        self, page_data: Page, html_data: Metadata, top_words: Dict[str, int]
+    ) -> None:
+        if self.client is None:
+            logger.error(f"Mongo connection not initialized")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)